### DISCUSSION TITLE
Meshdiag Topology not working

### DISCUSSION THREAD

**Discussion by HimanshuTembhe** [openthread/openthread#10346] [Q&A]
@jwhui , command meshdiag topology is not working for particular node , even though i have enabled it in the config file , it is showing an error . How to resolve it ?
![error1](https://github.com/openthread/openthread/assets/99601153/78c83de7-6978-4c14-9eec-85690fd65aad)
![error2](https://github.com/openthread/openthread/assets/99601153/f1770855-6dab-4c5d-adcf-e40e33ff264b)

**Comment by jwhui** [comment]
I don't think OTNS is designed to handle the `meshdiag topology` command. The command has a relatively long timeout (5 second default). I believe OTNS expects commands to finish more quickly than that.

**Comment by HimanshuTembhe** [comment]
Can we increase timeout for that particular command ?

**Reply by jwhui** [reply]
Does the `meshdiag topology` command work when you just have once device?

BTW, you can also reply to previous comments rather than suggesting new answers each time.

**Comment by EskoDijk** [comment]
Created an issue for OTNS2 to try the command and if it fails add the support for it: https://github.com/openthread/ot-ns/issues/704
In the current design, all commands that take more time to provide a response have to be placed on a special list of commands-with-delay.

**Reply by EskoDijk** [reply]
It looks like it works in current OTNS:

```
node 3> meshdiag topology
Started
id:32 rloc16:0x8000 ext-addr:a227c7da086357af ver:5
3-links:{ 49 }
id:49 rloc16:0xc400 ext-addr:4e989fcb28d32adb ver:5 - parent - leader
3-links:{ 32 }
Done
node 3>
```
